src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandLineHelpViewer.cs

```diff
@@ -0,0 +1,120 @@
+using Microsoft.DotNet.Cli.CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Cli.CommandParsing
+{
+    internal static class CommandLineHelpViewer
+    {
+        private static int columnGutterWidth = 3;
+        private const string templateArgument = "TEMPLATE";
+        internal static string HelpView(this Command newCommand)
+        {
+            var helpView = new StringBuilder();
+            WriteSynopsys(helpView, newCommand);
+            WriteArgumentsSection(helpView);
+            WriteOptionsSection(newCommand, helpView);
+            return helpView.ToString();
+        }
+
+        private static void WriteSynopsys(StringBuilder helpView, Command command)
+        {
+            helpView.AppendLine(string.Format(LocalizableStrings.CommandUsage, command.Name, templateArgument));
+            helpView.AppendLine();
+        }
+
+        private static void WriteArgumentsSection(StringBuilder helpView)
+        {
+            helpView.AppendLine(LocalizableStrings.ArgumentsSectionTitle);
+            WriteColumnizedSummary($"  <{templateArgument}>", string.Format(LocalizableStrings.ArgumentDescriptionTemplate, templateArgument), 20, helpView);
+            helpView.AppendLine();
+        }
+
+        private static void WriteOptionsSection(
+            Command command,
```

Comment:
minor: consider homogeneous parameter order.  In some peer methods, for the same parameters order is
- StringBuilder, Command
- here it is Command, StringBuilder